app/src/main/java/com/example/tmdb/data/HomeViewModel.kt

```diff
@@ -0,0 +1,57 @@
+package com.example.tmdb.data
+
+import androidx.lifecycle.ViewModel
+import com.example.tmdb.data.MovieDetails.Credits
+import com.example.tmdb.data.MovieDetails.Details
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+
+class HomeViewModel(
+    var movieRepository: MovieRepository,
+) : ViewModel(), KoinComponent {
+    fun getPopularMovies(): SharedFlow<List<MovieItemViewState>> {
+        return movieRepository.loadPopularMovies()
+    }
+
+    fun getNowPlayingMovies(): SharedFlow<List<MovieItemViewState>> {
+        return movieRepository.loadNowPlayingMovies()
+    }
+
+    fun getUpcomingMovies(): SharedFlow<List<MovieItemViewState>> {
+        return movieRepository.loadUpcomingMovies()
+    }
+
+    fun getTopRatedMovies(): SharedFlow<List<MovieItemViewState>> {
+        return movieRepository.loadTopRatedMovies()
+    }
+
+    fun addToFavorite(movieId: Int) {
+        val movieDetails = MutableStateFlow<Details?>(null)
+        val movieCredits = MutableStateFlow<Credits?>(null)
+
+        CoroutineScope(Dispatchers.IO).launch {
+            movieRepository.getMovieDetails(movieId).collect { movieDetails.emit(it) }
```

Comment:
You can use `.first()` operator to get first emitted element from the flow instead of `collect` all the items.